### PR TITLE
:sparkles: POST /api/preview

### PR DESCRIPTION
**Pull request by fabienjuif**

### CODE REVIEW

File: src/script/server/api/services/doc/docgen/docgen.spec.js [hunk 1]
```diff
@@ -126,14 +143,13 @@ describe('server/api/models/docgen', () => {
     }))
 
     // calls
-    await service.resolve('/foo/bar/test')
+    await service()(...toRedux('/foo/bar/test'))
 
     // asserts
-    expect(fs.readFileAsync.mock.calls.length).toBe(2)
+    expect(fs.readFileAsync.mock.calls.length).toBe(1)
```
Comment (fabienjuif) [inline src/script/server/api/services/doc/docgen/docgen.spec.js]: @bpetetot Do you have an idea about this ?

File: src/script/server/api/services/doc/docgen/docgen.spec.js [hunk 2]
```diff
@@ -144,10 +160,10 @@ describe('server/api/models/docgen', () => {
     }))
 
     // calls
-    await service.resolve('/foo/bar/test/index')
+    await service()(...toRedux('/foo/bar/test/index'))
 
     // asserts
-    expect(fs.readFileAsync.mock.calls.length).toBe(2)
+    expect(fs.readFileAsync.mock.calls.length).toBe(1)
```
Comment (fabienjuif) [inline src/script/server/api/services/doc/docgen/docgen.spec.js]: @bpetetot Do you have an idea about this ?

Comment (fabienjuif) [reply src/script/server/api/services/doc/docgen/docgen.spec.js]: Ok catch it ^^

File: src/script/server/api/services/doc/docgen/docgen.spec.js (hunk 1 above)
Comment (fabienjuif) [reply src/script/server/api/services/doc/docgen/docgen.spec.js]: Ok catch it ^^

**Comment by fabienjuif** [comment]
@bpetetot can you review this so I can finish gui impacts ?

File: src/script/server/api/interfaces/props.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const doc = require('../services/doc')
 const get = {
   method: 'get',
   path: 'props',
-  handler: async () => doc.generate('docgen'),
+  handler: async () => doc.docgen(),
```
Comment (bpetetot) [inline src/script/server/api/interfaces/props.js]: doesn't need to be async

File: src/script/server/api/interfaces/readme.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ const doc = require('../services/doc')
 const get = {
   method: 'get',
   path: 'readme',
-  handler: async () => doc.generate('markdown'),
+  handler: async () => doc.markdown(),
```
Comment (bpetetot) [inline src/script/server/api/interfaces/readme.js]: doesn't need to be async

File: src/script/server/api/services/doc/html/html.js [hunk 5]
```diff
@@ -2,10 +2,12 @@ const { State } = require('markup-it')
 const markdown = require('markup-it/lib/markdown')
 const html = require('markup-it/lib/html')
 
-module.exports = (source) => {
+module.exports = source => async () => {
```
Comment (bpetetot) [inline src/script/server/api/services/doc/html/html.js]: why is it async ?

**Review by bpetetot — CHANGES_REQUESTED** [review]

**Review by bpetetot — APPROVED** [review]
👌

**Pull request merged**